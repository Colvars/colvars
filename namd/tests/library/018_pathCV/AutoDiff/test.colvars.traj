# step         s                     r_s                    z                     r_z                    
           0   3.00814883534435e-215 3.00814883534435e-215   5.04223676928709e-01  5.04223676928709e-01  
           1   2.79773744065899e-215 -8.47753442854012e-07   5.04053981240470e-01  5.04215184762215e-01  
           2   2.60900632724628e-215  5.84013717452316e-06   5.04017885810743e-01  5.04206024707080e-01  
           3   2.48746158657393e-215  1.83444183792222e-05   5.03925949021255e-01  5.04194621361562e-01  
           4   2.33749398457775e-215  3.30663322771117e-05   5.03744457918909e-01  5.04179636036195e-01  
           5   2.19481225930835e-215  4.50102286359236e-05   5.03426140315685e-01  5.04168625648446e-01  
           6   2.06818775768437e-215  6.23252616516950e-05   5.02953441020839e-01  5.04160718764085e-01  
           7   1.99241404617557e-215  7.84867300879842e-05   5.02333036275841e-01  5.04157231816793e-01  
           8   1.94602382038457e-215  1.04106424125391e-04   5.01588199274176e-01  5.04151900677087e-01  
           9   1.90193758423791e-215  1.30553332245749e-04   5.00771453810440e-01  5.04145657103784e-01  
          10   1.88162886110026e-215  1.51318200440576e-04   4.99981821584820e-01  5.04130336462937e-01  
