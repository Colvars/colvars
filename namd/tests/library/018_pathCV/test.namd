# FILENAMES
set outName             [file rootname [file tail [info script]]]

# MD SETUP
timestep          0.5
numSteps		  10

# INPUT
structure               ../Common/da.psf 
parameters              ../Common/par_all22_prot.inp
paraTypeCharmm          on
coordinates             ../Common/da.min.pdb

temperature 250

# OUTPUT

outputname              ${outName}
outputenergies          100
outputtiming            2000
binaryoutput            no

seed		87654321


# CONSTANT-T
langevin                on
langevinTemp            300.0
langevinDamping         10.0
langevinHydrogen        on

# SPACE PARTITIONING
splitpatch              hydrogen
hgroupcutoff            2.8
stepspercycle           20
margin                  1.0

# CUT-OFFS
switching               on
switchdist              9.0
cutoff                  11.0
pairlistdist            12.0

# RESPA 
fullElectFrequency      1
nonbondedFreq           1

# 1-4 NON-BONDED
exclude                 scaled1-4
1-4scaling              1.0

# COM
commotion               no

# SHAKE
rigidBonds              none

# This does 2D extended-Lagrangian ABF in path coordinates (s, z)
# See Branduardi et al. JCP 2007 for details

# Parameters for the path colvars

namespace eval pathCV {
  set lambda     20.   ;# as in Branduardi et al.: should be 1/d^2 where d is the distance between images
  set tolerance  1e-4  ;# images whose exponential term is above this are included in the calculation
  set freq       5     ;# number of timesteps between two updates, where all images are enabled for one
                       ;# timestep to decide which ones should be disabled next
  set min_images 2     ;# if all exponential terms are small (moving away from the path), no fewer than
                       ;# this many images are retained
}
source                  ../../../../colvartools/pathCV.tcl

set atoms "atomNumbersRange 1-104"

set images { ../Common/da.xyz ../Common/310turn-ideal.xyz }
set nImages [llength $images]

set config "
colvarsTrajFrequency 1

colvar {
   name                  s

   width                 0.05
   lowerboundary        -0.2
   upperboundary         1.2
   lowerwallconstant     10.0
   upperwallconstant     10.0

   scriptedFunction      pathCVs
   extendedLagrangian    on
   extendedFluctuation   0.01
"

for {set i 0} {$i < $nImages} {incr i} {
  append config "rmsd {
     atoms {
       $atoms
     }
     refpositionsfile    [lindex $images $i]
   }
   "
}
append config "}"

append config "
colvar {
   name                  z

   width                 0.05
   lowerboundary        -0.5
   upperboundary         1.5

   scriptedFunction      pathCVz
   extendedLagrangian    on
   extendedFluctuation   0.01
"

for {set i 0} {$i < $nImages} {incr i} {
  append config "rmsd {
     atoms {
       $atoms
     }
     refpositionsfile    [lindex $images $i]
   }
   "
}
append config "}"

append config "
abf {
  colvars s z
  fullSamples 0
}"


colvars	on
cv config $config

