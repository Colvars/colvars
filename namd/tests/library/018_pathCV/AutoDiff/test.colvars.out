colvars: ----------------------------------------------------------------------
colvars: Please cite Fiorin et al, Mol Phys 2013:
colvars:  https://dx.doi.org/10.1080/00268976.2013.813594
colvars: in any publication based on this calculation.
colvars: SMP parallelism is enabled; if needed, use "smp off" to override this.
colvars: This version was built with the C++11 standard or higher.
colvars: The final output state file will be "test.colvars.state".
colvars: Opening trajectory file "test.colvars.traj".
colvars: Redefining the Tcl "cv" command to the new script interface.
colvars: ----------------------------------------------------------------------
colvars: Reading new configuration:
colvars: # units = "" [default]
colvars: # smp = on [default]
colvars: # colvarsTrajFrequency = 1
colvars: # colvarsRestartFrequency = 0 [default]
colvars: # scriptedColvarForces = off [default]
colvars: # scriptingAfterBiases = off [default]
colvars: ----------------------------------------------------------------------
colvars:   Initializing a new collective variable.
colvars:   # name = "s"
colvars:   Initializing a new "rmsd" component.
colvars:     # name = "" [default]
colvars:     # componentCoeff = 1 [default]
colvars:     # componentExp = 1 [default]
colvars:     # period = 0 [default]
colvars:     # wrapAround = 0 [default]
colvars:     # forceNoPBC = off [default]
colvars:     # scalable = on [default]
colvars:       Initializing atom group "atoms".
colvars:       # name = "" [default]
colvars:       # centerReference = off [default]
colvars:       # rotateReference = off [default]
colvars:       # atomsOfGroup = "" [default]
colvars:       # indexGroup = "" [default]
colvars:       # psfSegID =  [default]
colvars:       # atomsFile = "" [default]
colvars:       # dummyAtom = ( 0 , 0 , 0 ) [default]
colvars:       # enableForces = on [default]
colvars:       # enableFitGradients = on [default]
colvars:       # printAtomIDs = off [default]
colvars:       Atom group "atoms" defined with 104 atoms requested: total mass = 727.821, total charge = 6.70552e-08.
colvars:     # refPositions =  [default]
colvars:     # refPositionsFile = "../Common/da.xyz"
colvars:     # refPositionsCol = "" [default]
colvars:     Warning: beginning from 2019-11-26 the XYZ file reader assumes Angstrom units.
colvars:     Enabling "centerReference" and "rotateReference", to minimize RMSD before calculating it as a variable: if this is not the desired behavior, disable them explicitly within the "atoms" block.
colvars:     This is a standard minimum RMSD, derivatives of the optimal rotation will not be computed as they cancel out in the gradients.
colvars:   Initializing a new "rmsd" component.
colvars:     # name = "" [default]
colvars:     # componentCoeff = 1 [default]
colvars:     # componentExp = 1 [default]
colvars:     # period = 0 [default]
colvars:     # wrapAround = 0 [default]
colvars:     # forceNoPBC = off [default]
colvars:     # scalable = on [default]
colvars:       Initializing atom group "atoms".
colvars:       # name = "" [default]
colvars:       # centerReference = off [default]
colvars:       # rotateReference = off [default]
colvars:       # atomsOfGroup = "" [default]
colvars:       # indexGroup = "" [default]
colvars:       # psfSegID =  [default]
colvars:       # atomsFile = "" [default]
colvars:       # dummyAtom = ( 0 , 0 , 0 ) [default]
colvars:       # enableForces = on [default]
colvars:       # enableFitGradients = on [default]
colvars:       # printAtomIDs = off [default]
colvars:       Atom group "atoms" defined with 104 atoms requested: total mass = 727.821, total charge = 6.70552e-08.
colvars:     # refPositions =  [default]
colvars:     # refPositionsFile = "../Common/310turn-ideal.xyz"
colvars:     # refPositionsCol = "" [default]
colvars:     Enabling "centerReference" and "rotateReference", to minimize RMSD before calculating it as a variable: if this is not the desired behavior, disable them explicitly within the "atoms" block.
colvars:     This is a standard minimum RMSD, derivatives of the optimal rotation will not be computed as they cancel out in the gradients.
colvars:   All components initialized.
colvars:   This colvar uses scripted function "pathCVs".
colvars:   # scriptedFunctionType = "scalar" [default]
colvars:   Expecting colvar value of type scalar number
colvars:   Sorted list of components for this scripted colvar:
colvars:   1 rmsd0001
colvars:   2 rmsd0002
colvars:   # period = 0 [default]
colvars:   # timeStepFactor = 1 [default]
colvars:   # width = 0.05
colvars:   # lowerBoundary = -0.2
colvars:   # upperBoundary = 1.2
colvars:   Reading legacy options lowerWall and lowerWallConstant: consider using a harmonicWalls restraint (caution: force constant would then be scaled by width^2).
colvars:   # lowerWall = -0.2
colvars:   Reading legacy options upperWall and upperWallConstant: consider using a harmonicWalls restraint (caution: force constant would then be scaled by width^2).
colvars:   # upperWall = 1.2
colvars:   Generating a new harmonicWalls bias for compatibility purposes.
colvars:   # hardLowerBoundary = off [default]
colvars:   # hardUpperBoundary = off [default]
colvars:   # expandBoundaries = off [default]
colvars:   # extendedLagrangian = on
colvars:   Enabling the extended Lagrangian term for colvar "s".
colvars:   # extendedTemp = 300 [default]
colvars:   # extendedFluctuation = 0.01
colvars:   Computed extended system force constant: 5961.57 [E]/U^2
colvars:   # extendedTimeConstant = 200 [default]
colvars:   Computed fictitious mass: 6.04034e+06 [E]/(U/fs)^2   (U: colvar unit)
colvars:   # outputEnergy = off [default]
colvars:   # extendedLangevinDamping = 1 [default]
colvars:   # reflectingLowerBoundary = off [default]
colvars:   # reflectingUpperBoundary = off [default]
colvars:   # outputValue = on [default]
colvars:   # outputVelocity = off [default]
colvars:   # outputTotalForce = off [default]
colvars:   # outputAppliedForce = off [default]
colvars:   # subtractAppliedForce = off [default]
colvars:   # runAve = off [default]
colvars:   # corrFunc = off [default]
colvars: ----------------------------------------------------------------------
colvars:   Initializing a new collective variable.
colvars:   # name = "z"
colvars:   Initializing a new "rmsd" component.
colvars:     # name = "" [default]
colvars:     # componentCoeff = 1 [default]
colvars:     # componentExp = 1 [default]
colvars:     # period = 0 [default]
colvars:     # wrapAround = 0 [default]
colvars:     # forceNoPBC = off [default]
colvars:     # scalable = on [default]
colvars:       Initializing atom group "atoms".
colvars:       # name = "" [default]
colvars:       # centerReference = off [default]
colvars:       # rotateReference = off [default]
colvars:       # atomsOfGroup = "" [default]
colvars:       # indexGroup = "" [default]
colvars:       # psfSegID =  [default]
colvars:       # atomsFile = "" [default]
colvars:       # dummyAtom = ( 0 , 0 , 0 ) [default]
colvars:       # enableForces = on [default]
colvars:       # enableFitGradients = on [default]
colvars:       # printAtomIDs = off [default]
colvars:       Atom group "atoms" defined with 104 atoms requested: total mass = 727.821, total charge = 6.70552e-08.
colvars:     # refPositions =  [default]
colvars:     # refPositionsFile = "../Common/da.xyz"
colvars:     # refPositionsCol = "" [default]
colvars:     Enabling "centerReference" and "rotateReference", to minimize RMSD before calculating it as a variable: if this is not the desired behavior, disable them explicitly within the "atoms" block.
colvars:     This is a standard minimum RMSD, derivatives of the optimal rotation will not be computed as they cancel out in the gradients.
colvars:   Initializing a new "rmsd" component.
colvars:     # name = "" [default]
colvars:     # componentCoeff = 1 [default]
colvars:     # componentExp = 1 [default]
colvars:     # period = 0 [default]
colvars:     # wrapAround = 0 [default]
colvars:     # forceNoPBC = off [default]
colvars:     # scalable = on [default]
colvars:       Initializing atom group "atoms".
colvars:       # name = "" [default]
colvars:       # centerReference = off [default]
colvars:       # rotateReference = off [default]
colvars:       # atomsOfGroup = "" [default]
colvars:       # indexGroup = "" [default]
colvars:       # psfSegID =  [default]
colvars:       # atomsFile = "" [default]
colvars:       # dummyAtom = ( 0 , 0 , 0 ) [default]
colvars:       # enableForces = on [default]
colvars:       # enableFitGradients = on [default]
colvars:       # printAtomIDs = off [default]
colvars:       Atom group "atoms" defined with 104 atoms requested: total mass = 727.821, total charge = 6.70552e-08.
colvars:     # refPositions =  [default]
colvars:     # refPositionsFile = "../Common/310turn-ideal.xyz"
colvars:     # refPositionsCol = "" [default]
colvars:     Enabling "centerReference" and "rotateReference", to minimize RMSD before calculating it as a variable: if this is not the desired behavior, disable them explicitly within the "atoms" block.
colvars:     This is a standard minimum RMSD, derivatives of the optimal rotation will not be computed as they cancel out in the gradients.
colvars:   All components initialized.
colvars:   This colvar uses scripted function "pathCVz".
colvars:   # scriptedFunctionType = "scalar" [default]
colvars:   Expecting colvar value of type scalar number
colvars:   Sorted list of components for this scripted colvar:
colvars:   1 rmsd0001
colvars:   2 rmsd0002
colvars:   # period = 0 [default]
colvars:   # timeStepFactor = 1 [default]
colvars:   # width = 0.05
colvars:   # lowerBoundary = -0.5
colvars:   # upperBoundary = 1.5
colvars:   # hardLowerBoundary = off [default]
colvars:   # hardUpperBoundary = off [default]
colvars:   # expandBoundaries = off [default]
colvars:   # extendedLagrangian = on
colvars:   Enabling the extended Lagrangian term for colvar "z".
colvars:   # extendedTemp = 300 [default]
colvars:   # extendedFluctuation = 0.01
colvars:   Computed extended system force constant: 5961.57 [E]/U^2
colvars:   # extendedTimeConstant = 200 [default]
colvars:   Computed fictitious mass: 6.04034e+06 [E]/(U/fs)^2   (U: colvar unit)
colvars:   # outputEnergy = off [default]
colvars:   # extendedLangevinDamping = 1 [default]
colvars:   # reflectingLowerBoundary = off [default]
colvars:   # reflectingUpperBoundary = off [default]
colvars:   # outputValue = on [default]
colvars:   # outputVelocity = off [default]
colvars:   # outputTotalForce = off [default]
colvars:   # outputAppliedForce = off [default]
colvars:   # subtractAppliedForce = off [default]
colvars:   # runAve = off [default]
colvars:   # corrFunc = off [default]
colvars: ----------------------------------------------------------------------
colvars: Collective variables initialized, 2 in total.
colvars: ----------------------------------------------------------------------
colvars:   Initializing a new "abf" instance.
colvars:   # name = "abf1" [default]
colvars:   # colvars = { s, z }
colvars:   # stepZeroData = off [default]
colvars:   # outputEnergy = off [default]
colvars:   # outputFreq = 0 [default]
colvars:   # timeStepFactor = 1 [default]
colvars:   # applyBias = on [default]
colvars:   # updateBias = on [default]
colvars:   # hideJacobian = off [default]
colvars:   Jacobian (geometric) forces will be included in reported free energy gradients.
colvars:   # fullSamples = 0
colvars:   # inputPrefix =  [default]
colvars:   # historyFreq = 0 [default]
colvars:   # shared = off [default]
colvars:   # maxForce =  [default]
colvars:   # CZARestimator = on [default]
colvars:   # integrate = on [default]
colvars:   # pABFintegrateFreq = 0 [default]
colvars:   # integrateInitMaxIterations = 10000 [default]
colvars:   # integrateInitTol = 1e-06 [default]
colvars:   # integrateMaxIterations = 100 [default]
colvars:   # integrateTol = 0.0001 [default]
colvars:   # UIestimator = off [default]
colvars:   Finished ABF setup.
colvars: ----------------------------------------------------------------------
colvars: Collective variables biases initialized, 1 in total.
colvars: # units = "" [default]
colvars: # smp = on [default]
colvars: # colvarsTrajFrequency = 1 [default]
colvars: # colvarsRestartFrequency = 0 [default]
colvars: # scriptedColvarForces = off [default]
colvars: # scriptingAfterBiases = off [default]
colvars: ----------------------------------------------------------------------
colvars: Collective variables initialized, 2 in total.
colvars: ----------------------------------------------------------------------
colvars:   Initializing a new "harmonicwalls" instance.
colvars:   # name = "sw"
colvars:   # colvars = { s }
colvars:   # stepZeroData = off [default]
colvars:   # outputEnergy = off [default]
colvars:   # outputFreq = 0 [default]
colvars:   # timeStepFactor = 1
colvars:   # writeTISamples = off [default]
colvars:   # writeTIPMF = off [default]
colvars:   # forceConstant = 1 [default]
colvars:   # targetForceConstant = -1 [default]
colvars:   # lowerWalls = { -0.2 }
colvars:   # upperWalls = { 1.2 }
colvars:   # lowerWallConstant = 0.025
colvars:   # upperWallConstant = 0.025
colvars:   The lower wall force constant for colvar "s" will be rescaled to 10 according to the specified width (0.05).
colvars:   The upper wall force constant for colvar "s" will be rescaled to 10 according to the specified width (0.05).
colvars: ----------------------------------------------------------------------
colvars: Collective variables biases initialized, 2 in total.
colvars: ----------------------------------------------------------------------
colvars: Collective variables module (re)initialized.
colvars: ----------------------------------------------------------------------
colvars: Updating NAMD interface:
colvars: updating atomic data (104 atoms).
colvars: updating group data (0 scalable groups, 0 atoms in total).
colvars: Updating NAMD interface:
colvars: updating atomic data (104 atoms).
colvars: updating group data (0 scalable groups, 0 atoms in total).
colvars: Re-initialized atom group for variable "s":0/0. 104 atoms: total mass = 727.821, total charge = 6.70552e-08.
colvars: Re-initialized atom group for variable "s":1/0. 104 atoms: total mass = 727.821, total charge = 6.70552e-08.
colvars: Re-initialized atom group for variable "z":0/0. 104 atoms: total mass = 727.821, total charge = 6.70552e-08.
colvars: Re-initialized atom group for variable "z":1/0. 104 atoms: total mass = 727.821, total charge = 6.70552e-08.
colvars: The final output state file will be "test.colvars.state".
colvars: PathCV WARNING: less than 2 RMSDs are above threshold, selecting 2 closest images
colvars: PathCV STEP 1   next pathCV flags  1 1
colvars:   Prepared sample and gradient buffers at step 0.
colvars: PathCV WARNING: less than 2 RMSDs are above threshold, selecting 2 closest images
colvars: PathCV STEP 6   next pathCV flags  1 1
colvars: PathCV WARNING: less than 2 RMSDs are above threshold, selecting 2 closest images
colvars: PathCV STEP 11   next pathCV flags  1 1
colvars: Saving collective variables state to "test.colvars.state".
colvars:   Integrated in 143 steps, error: 9.52755e-07
colvars:   Integrated in 143 steps, error: 8.74992e-07
